### ISSUE TITLE
Profile only picking up only mobs/entities

### ISSUE THREAD

**Issue by AussieWarGod** [MinimallyCorrect/TickProfiler#57]
We are running Tickprofiler on a Hermitpack server and are finding that it doesn't pick up on laggy blocks at all for some reason. 

provided are a Sponge timing report and a profile for reference, taken at the same time.

https://timings.aikar.co/v2/?id=e91dc6eec42e48d28e8f120ea3ec9bb0

http://i.imgur.com/oSEyqaO.png
http://i.imgur.com/GiOpd06.png

**Comment by LunNova** [comment]
Please upload a log file, not enough info there to determine the cause

**Comment by AussieWarGod** [comment]
Going through the logs i see nothing of use, is there a debug mode i could use to get more info?

the best i can give you is that it lags the server a bit when a profile is run, but that is normal and happens on our 1.7.10 servers as well.

What i have noticed is the only things it shows are vanilla mobs/item entities on the ground, whether the tps is at 20 or 0.4.

As you can see via the provided timing report there was obviously a quarry causing lag somewhere and there was also (we know for a fact) various beehives causing lag as they chunk load on the current version of HP. 

I have no idea about anything to do with coding or how tickprofiler works but is there any chance it could not be hooking in to something to do with modded items/blocks/mobs?

**Comment by LunNova** [comment]
Post the full logs. Don't look at them yourself.

(if I ask for logs, this means I need the logs. Replying with other unrelated information is not actually helpful.)

**Comment by AussieWarGod** [comment]
Log, as requested, this was at the time of my running the profile and sponge timings provided above. I did warn you it's contents are less than useful. I did ask if there was some sort of debugging mode i could use. 

to clarify, tickprofiler doesn't say anything is wrong with it or throw errors, it just doesn't seem to pick up non-vanilla blocks or entities.

http://pastebin.com/TWBdM63g

**Comment by AussieWarGod** [comment]
if you can let me know a way i can provide more detail, or if there is any more information you need please let me know and i will provide it to you ASAP.

I looked through sponge configs and server properties and could find nothing that might make the log files more detailed.

**Comment by LunNova** [comment]
Your log file's missing any [TickProfiler] or [ModPatcher] entries. Make sure you got the right file - fml-server-latest.log or fml-server-x.log would be appropriate, a console log or other log doesn't include everything.

If that was the FML log, it's messed up and missing most non-vanilla entries for some reason.

**Comment by AussieWarGod** [comment]
Timing report
https://timings.aikar.co/v2/?id=831515574b4c472e8f481caaabd75c22

Profile
http://i.imgur.com/KAhjOaU.png
http://i.imgur.com/malNh6H.png

Full FML Log
[fml-server-latest.txt](https://github.com/nallar/TickProfiler/files/566248/fml-server-latest.txt)

**Comment by AussieWarGod** [comment]
the profile should be towards the end of the logs, i pulled it shortly after i ran the profile and the timing report. 

tps is almost 20 obviously so it's not the ideal situation for a demonstration but you can easily see the differences between timings from the first timings/screenshots i posted.

Sorry about the late replies, i work a lot and the timezone difference makes it difficult

**Comment by LunNova** [comment]
Looks like the problem is an interaction between Sponge's mixin system and ModPatcher.

Their transformer tries its hardest to be the last transformer to run. Unfortunately this means applying a patch after their patches is impossible. I had a workaround for it but it looks like something's changed and that's no longer working correctly, either due to a Sponge change or an interaction between something else in your environment, modpatcher and sponge.

Can you add `-Dnallar.LaunchClassLoaderUtil.dumpTransformers=true` to your java options to enable dumping of transformers in the logs, then upload a new log? This would help with diagnosing what's gone wrong.

It may take me a while to resolve this one, transformer and classloading related issues are a massive pain. :'(

**Comment by axle2005** [comment]
[fml-server-3.txt](https://github.com/nallar/TickProfiler/files/567632/fml-server-3.txt)
This was loaded with those parameter's, not really seeing much different between it and another, so hopefully thats what you were looking for.

**Comment by LunNova** [comment]
```
[16:30:08] [main/DEBUG] [mixin/]: Adding new mixin transformer proxy #3
[16:30:08] [main/INFO] [ModPatcher/]: Transformers: [net.minecraftforge.fml.common.asm.transformers.PatchingTransformer@5109e8cf, org.spongepowered.asm.mixin.transformer.MixinTransformer$Proxy@3f672204, TransformerWrapper(net.minecraftforge.fml.common.asm.transformers.BlamingTransformer, FMLCorePlugin), TransformerWrapper(net.minecraftforge.fml.common.asm.transformers.SideTransformer, FMLCorePlugin), TransformerWrapper(net.minecraftforge.fml.common.asm.transformers.EventSubscriptionTransformer, FMLCorePlugin), TransformerWrapper(net.minecraftforge.fml.common.asm.transformers.EventSubscriberTransformer, FMLCorePlugin), TransformerWrapper(invtweaks.forge.asm.ContainerTransformer, FMLPlugin), TransformerWrapper(li.cil.oc.common.asm.ClassTransformer, TransformerLoader), TransformerWrapper(vazkii.quark.base.asm.ClassTransformer, LoadingPlugin), TransformerWrapper(org.spongepowered.common.launch.transformer.SpongeSuperclassTransformer, SpongeCoremod), net.minecraftforge.fml.common.asm.transformers.DeobfuscationTransformer@78b41097, net.minecraftforge.fml.common.asm.transformers.AccessTransformer@2c2db130, invtweaks.forge.asm.ITAccessTransformer@327c7bea, org.spongepowered.mod.asm.transformer.SpongeAccessTransformer@348d18a3, net.minecraftforge.fml.common.asm.transformers.ModAccessTransformer@6c65860d, net.minecraftforge.fml.common.asm.transformers.ItemStackTransformer@2d000e80, org.spongepowered.asm.mixin.transformer.MixinTransformer$Proxy@7cf283e1, TransformerWrapper(net.blay09.mods.defaultoptions.coremod.DefaultOptionsClassTransformer, Default Options), TransformerWrapper(lumien.randomthings.asm.ClassTransformer, LoadingPlugin), TransformerWrapper(shetiphian.core.asm.ClassTransformer, ShetiPhian-ASM), TransformerWrapper(net.fybertech.nwr.NWRTransformer, NWRTweak), TransformerWrapper(team.chisel.common.asm.ChiselTransformer, ChiselCorePlugin), TransformerWrapper(com.enderio.core.common.transform.EnderCoreTransformer, EnderCorePlugin), net.minecraftforge.fml.common.asm.transformers.TerminalTransformer@3506d826, org.spongepowered.asm.mixin.transformer.MixinTransformer$Proxy@35dd9ed3, me.nallar.modpatcher.ModPatcherTransformer$ClassTransformer@20e6c4dc]
java.lang.Throwable
    at me.nallar.modpatcher.LaunchClassLoaderUtil.dumpTransformersIfEnabled(LaunchClassLoaderUtil.java:78) [ModPatcher.jlib:?]
    at me.nallar.modpatcher.ModPatcherTweaker.getLaunchArguments(ModPatcherTweaker.java:54) [ModPatcher.jlib:?]
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:126) [launchwrapper-1.12.jar:?]
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_101]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_101]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_101]
    at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_101]
    at net.minecraftforge.fml.relauncher.ServerLaunchWrapper.run(ServerLaunchWrapper.java:62) [hp.jar:?]
    at net.minecraftforge.fml.relauncher.ServerLaunchWrapper.main(ServerLaunchWrapper.java:31) [hp.jar:?]
```

Good news: Confirms the problem isn't with making sure our transformer runs after the mixin transformer, that part worked correctly. So it's not a transformer order issue! :)

Bad news: That means I'm not sure what's broken.

Busy right now due to the aftermath of a house move, I'll hopefully have time to investigate more thoroughly on Hermitpack with Sponge soon.

**Comment by AussieWarGod** [comment]
Should i run the same on our RR4 test server to see if we get the same results

**Comment by AussieWarGod** [comment]
Can confirm this issue persists over modpacks, 
Infinity Lite
Resonant Rise 4
Infinity Lite 

Are all affected so far

**Comment by AussieWarGod** [comment]
DireWolf 1.10 is also affected.

At this point i can guarantee this is a sponge issue, any luck finding the cause?

**Comment by AussieWarGod** [comment]
any news on this nallar?

**Comment by LunNova** [comment]
Should be fixed in `TickProfiler-1.10.2.jenkins.13`. Tested with `spongeforge-1.10.2-2171-5.1.0-BETA-1963.jar` and `forge-1.10.2-12.18.3.2185-universal.jar`.

**Comment by JaSpr** [comment]
We appear to be having this same issue in 1.12.2 but on a non-sponge server.  I updated to TickProfiler-1.12-0.0.2 but it still only outputs results for mobs and entities.
